src/unity/toolkits/one_shot_object_detection/one_shot_object_detector.cpp

```diff
@@ -0,0 +1,171 @@
+    phi_ = 0.0;
+    gamma_ = 0.0;
+    std::normal_distribution<double> focal_distribution((double)width_, focal_stdev_);
+    focal_generator_.seed(seed+6);
+    focal_ = focal_distribution(focal_generator_);
+    dz_ = focal_;
+  }
+
+private:
+  int width_;
+  int height_;
+  double focal_stdev_ = 40.0;
+  std::default_random_engine focal_generator_;
+  double theta_;
+  double phi_;
+  double gamma_;
+  int dz_;
+  double focal_;
+
+};
+
+gl_sframe _augment_data(gl_sframe data, gl_sframe backgrounds, long seed) {
+  // TODO: Get input image from the data sframe.
+  // TODO: Use backgrounds from the background SFrame.
+  // TODO: Generalize 1024 and 676 to be the width and height of the image 
+  //       passed in.
+  ParameterSweep parameter_sampler = ParameterSweep(2*1024, 2*676);
+  // TODO: Take n as input.
+  int n = 1;
+  for (int i = 0; i < n; i++) {
+    parameter_sampler.sample(seed+i);
+
+    Matrix3f mat = get_transformation_matrix(2*1024, 2*676,
+      parameter_sampler.get_theta(),
+      parameter_sampler.get_phi(),
+      parameter_sampler.get_gamma(),
+      1024/2,
+      676/2,
+      parameter_sampler.get_dz(),
+      parameter_sampler.get_focal());
+
+    // TODO: Construct annotations that conform to the OD format, from the 
+    // four warped corner points.
+
+    rgb8_image_t starter_image, background;
+    // TODO: Don't hardcode this.
+    read_image("in-affine.jpg", starter_image, jpeg_tag());
+    // TODO: Don't hardcode this. Fetch this from the backgrounds SFrame in a 
+    //       loop.
+    read_image("background.jpg", background, jpeg_tag());
+
+    matrix3x3<double> M(mat.inverse());
+    // TODO: Use a background.
+    // TODO: Use a mask during superposition on a random background.
+    rgb8_image_t transformed(rgb8_image_t::point_t(view(starter_image).dimensions()*2));
+    fill_pixels(view(transformed),rgb8_pixel_t(255, 255, 255));
+    resample_pixels(const_view(starter_image), view(transformed), M, bilinear_sampler());
+
+    // TODO: Write these images into an SFrame that this function can return.
+    std::string output_filename = "out-perspective-" + std::to_string(i) + ".jpg";
+    write_view(output_filename, view(transformed), jpeg_tag());
+  }
+  // TODO: Return the augmented data once the SFrame is written.
+  return data;
+}
+
+one_shot_object_detector::one_shot_object_detector() {
+  model_.reset(new turi::object_detection::object_detector());
+}
+
+gl_sframe one_shot_object_detector::augment(gl_sframe data,
+                                            std::string target_column_name,
+                                            gl_sframe backgrounds,
+                                            long seed){
+  gl_sframe augmented_data = _augment_data(data, backgrounds, seed);
+  // TODO: Call object_detector::train from here once we incorporate mxnet into
+  // the C++ Object Detector.
+  return augmented_data;
+}
+/* TODO: We probably don't need `evaluate` and `export_to_coreml` on the C++ 
```

Comment:
Let's just add them when we actually need them. There's no need to stub out functions that aren't required by some interface

Reply:
Done.